core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitMutationOperationChecker.java

```diff
@@ -0,0 +1,96 @@
+package com.scalar.db.transaction.consensuscommit;
+
+import com.scalar.db.api.ConditionalExpression;
+import com.scalar.db.api.Delete;
+import com.scalar.db.api.DeleteIf;
+import com.scalar.db.api.DeleteIfExists;
+import com.scalar.db.api.Mutation;
+import com.scalar.db.api.MutationCondition;
+import com.scalar.db.api.Operation;
+import com.scalar.db.api.Put;
+import com.scalar.db.api.PutIf;
+import com.scalar.db.api.PutIfExists;
+import com.scalar.db.api.PutIfNotExists;
+import com.scalar.db.exception.storage.ExecutionException;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
+public class ConsensusCommitMutationOperationChecker {
+
+  private final TransactionTableMetadataManager transactionTableMetadataManager;
+
+  public ConsensusCommitMutationOperationChecker(
+      TransactionTableMetadataManager transactionTableMetadataManager) {
+    this.transactionTableMetadataManager = transactionTableMetadataManager;
+  }
+
+  private TransactionTableMetadata getTableMetadata(Operation operation) throws ExecutionException {
+    TransactionTableMetadata metadata =
+        transactionTableMetadataManager.getTransactionTableMetadata(operation);
+    if (metadata == null) {
+      throw new IllegalArgumentException(
+          "The specified table is not found: " + operation.forFullTableName().get());
+    }
+    return metadata;
+  }
+
+  public void check(Mutation mutation) throws ExecutionException {
+    if (mutation instanceof Put) {
+      check((Put) mutation);
+    } else if (mutation instanceof Delete) {
+      check((Delete) mutation);
+    }
+  }
+
+  private void check(Put put) throws ExecutionException {
+    put.getCondition()
+        .ifPresent(
+            condition -> {
+              if (!(condition instanceof PutIf
+                  || condition instanceof PutIfNotExists
+                  || condition instanceof PutIfExists)) {
+                throw new IllegalArgumentException("The condition is not allowed on Put operation");
```

Comment:
I guess a user would be interested in the type of the wrong `condition` when receiving this exception. How about showing it in the error message?

Reply:
Yes, that is a good point, thank you. I will add it.

Reply:
Revised in https://github.com/scalar-labs/scalardb/pull/873/commits/4e05db8de6de4d94bc81de22ad3c3b40bfd7a321